src/pages/ProductDetail/ProductDetail.js

```diff
@@ -1,7 +1,285 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import * as thousand from '../../styles/thousand';
+import theme from '../../styles/theme';
+import {
+  faHeart,
+  faCartShopping,
+  faGift,
+  faBookOpen,
+} from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import PaymentModal from './PaymentModal';
+import StarRating from './StarRating';
 
 const ProductDetail = () => {
-  return <h1>hi</h1>;
+  const [book, setBook] = useState([]);
+
+  useEffect(() => {
+    fetch('/data/detail.json')
+      .then(res => res.json())
+      .then(data => setBook(data));
+  }, []);
+
+  const [isOpen, setOpen] = useState(false);
+
+  const handleClick = () => {
+    setOpen(true);
+  };
+
+  const handleModalSubmit = () => setOpen(false);
+
+  const handleModalCancel = () => setOpen(false);
```

Comment:
```suggestion
  const closeModal = () => setOpen(false);
```
각각의 함수를 사용하기 보다는 하나의 함수를 만들고,  그 함수를 필요한 곳에 사용해주세요~